src/provider.coffee

```diff
@@ -126,6 +127,9 @@ class Provider
     @context_label  = @body.context_label
     @context_title  = @body.context_title
 
+    # Load up the extensions!
+    extension.init(@) for extension_name, extension of extensions
```

Comment:
So oddly enough after this line an extension persists across all UNIQUE Provider objects that I create for testing. Here's a small test class I had set up, ContentExtension.coffee. For some reason the 2nd test case fails because it is referencing data from the first unrelated provider object? Maybe you have more insight @omsmith?

```js
lti = require '../'


describe 'LTI.Extensions.Content', () ->
  describe 'Initialization', () ->
    it 'should initialize an object if ext_content_return_types is set', () ->
      provider = new lti.Provider 'key', 'secret'

      req =
        body:
          ext_content_return_types: 'file,iframe'

      provider.parse_request req
      provider.ext_content.should.exist

    it 'should be false if ext_content_return_types is missing', () ->
      provider = new lti.Provider 'key', 'secret'

      req = { body: {} }

      provider.parse_request req
      provider.ext_content.should.equal false
```